apps/aehttp/src/aehttp_helpers.erl

```diff
@@ -288,3 +308,57 @@ compute_contract_call_data() ->
                 {error, {400, [], #{<<"reason">> => Reason}}}
         end
     end.
+
+get_transaction(TxKey, TxStateKey) ->
+    fun(_Req, State) ->
+        TxHash = maps:get(TxKey, State),
+        case aec_db:read_tx(TxHash) of
+            [] ->
+                {error, {404, [], #{<<"reason">> => <<"Transaction not found">>}}};
+            [{Block0, Tx}] ->
+                BlockHash =
+                    case Block0 of
+                        mempool -> mempool;
+                        BH when is_binary(BH) -> BH
+                    end,
+                {ok, maps:put(TxStateKey, #{tx => Tx,
+                                            tx_block_hash => BlockHash},
+                             State)}
+        end
+    end.
+
+parse_tx_encoding(TxEncodingKey) ->
+    fun(_Req, State) ->
+        TxEncoding =
+            case maps:get(TxEncodingKey, State) of
+                message_pack -> message_pack;
```

Comment:
Can these clauses with atom be deleted? If I understand the flow, these atoms would have been generated dynamically by Swagger decoding JSON and we want to ensure during testing that it is not the case. (I am concerned about potentially dynamically generated atoms.)

Reply:
This is the product of swagger parsing enum-ed types. If a parameter is set to be contained in a certain interval of values - swagger [does parse it to an **exsisting** atom](https://github.com/aeternity/epoch/blob/master/apps/aehttp/src/swagger/swagger_api.erl#L1641) (so no risk of dynamically generating atoms).
What swagger does not do is actually validating this parsed atom to be part of the enumed ones. This is something we're doing by ourselves.